### ISSUE TITLE
[BUG][cpp-restsdk] Enums not generated

### ISSUE THREAD

**Issue by gluttony38** [OpenAPITools/openapi-generator#8937]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

A member described with an enum has no corresponding enum when generating code for cpp-restsdk (and it seems to be the same for any C++ generators), it is then hard to map possible values from server to our code, and to ensure not setting an invalid value.

##### openapi-generator version

5.0.1

##### OpenAPI declaration file content or url

https://github.com/OpenAPITools/openapi-generator/blob/master/samples/yaml/pet.yml

Pet status is described with an enum:

```
status: 
  type: string
  description: pet status in the store
  enum: 
    - available
    - pending
    - sold
```

##### Generation Details

Nothing special, but as you can see I use your sample that has the same issue.

##### Steps to reproduce

Simply generate the code for your sample, you can see in Java there is an enum generated
(see https://github.com/OpenAPITools/openapi-generator/blob/master/samples/client/petstore/java/resttemplate/src/main/java/org/openapitools/client/model/Pet.java):

```
  public enum StatusEnum {
    AVAILABLE("available"),
    
    PENDING("pending"),
    
    SOLD("sold");

    private String value;

    StatusEnum(String value) {
      this.value = value;
    }
    [...]
  }
  [...]
  private StatusEnum status;
```

But in C++ a direct string is used making it hard to manage cleanly such member (see https://github.com/OpenAPITools/openapi-generator/blob/master/samples/client/petstore/cpp-restsdk/client/model/Pet.h):
`utility::string_t m_Status`

##### Related issues/PRs

Found issues for another generator but not for any C++ one.

##### Suggest a fix

Like in Java, instead of simply using a string, generate an enum and use it as type for such members.

**Comment by xconverge** [comment]
I was just about to open this issue for the cpp-QT5 generator, tuning in here to see what gets discussed

**Comment by etherealjoy** [comment]
The issue is there for all `C++` generators.
The enum is generated if the top level property is enum, but not generated if it is part of another property.

generated in this case, probably missing only for one or two `c++` generators
```yaml
definitions:
status: 
  type: string
  description: pet status in the store
  enum: 
    - available
    - pending
    - sold
```
not generated here
```yaml
definitions:
property1:
  type: object
  properties:
    status: 
      type: string
      description: pet status in the store
      enum: 
        - available
        - pending
        - sold
```

We need to tackle this case too

**Comment by saterHATER** [comment]
Can confirm I'm stuck on this issue as well. If you're willing to use your own templates, you might be able to use 5.0.1 without major issues for cpp-restsdk. It's unfortunate I can't use @wing328's commits he made just for me :(

**Comment by wing328** [comment]
Merged #19919 

Please give it a try with the latest master or snapshot version

Thanks @SaverioCode for the PR.
